contracts/rewards/RewardsController.sol

```diff
@@ -227,24 +227,30 @@ contract RewardsController is RewardsDistributor, VersionedInitializable, IRewar
     if (amount == 0) {
       return 0;
     }
-    uint256 unclaimedRewards = _usersUnclaimedRewards[user][reward];
+    uint256 totalRewards;
 
-    if (amount > unclaimedRewards) {
-      _distributeRewards(user, _getUserStake(assets, user));
-      unclaimedRewards = _usersUnclaimedRewards[user][reward];
+    _distributeRewards(user, _getUserStake(assets, user));
+    for (uint256 i = 0; i < assets.length; i++) {
+      address asset = assets[i];
+      totalRewards += _assets[asset].rewards[reward].usersData[user].accrued;
+
+      if (totalRewards <= amount) {
+        _assets[asset].rewards[reward].usersData[user].accrued = 0;
+      } else {
+        uint256 difference = totalRewards - amount;
+        _assets[asset].rewards[reward].usersData[user].accrued -= uint128(difference);
+        break;
+      }
     }
 
-    if (unclaimedRewards == 0) {
+    if (totalRewards == 0) {
       return 0;
     }
 
-    uint256 amountToClaim = amount > unclaimedRewards ? unclaimedRewards : amount;
-    _usersUnclaimedRewards[user][reward] = unclaimedRewards - amountToClaim; // Safe due to the previous line
+    _transferRewards(to, reward, totalRewards);
```

Comment:
Don't believe it should be `totalRewards` that is transferred, the natspec also states that `amount` is the amount to claim.

Reply:
if you look at the old code, `amountToClaim` was transferred, which is the adjusted amount to claim against the user total rewards (line 241 of the removed diff). So caller can pass an amount that is bigger than the total rewards to signal the need to claim everything. In this case, the `if(totalRewards <= amount)` mimics the amount adjustment, so that if amount < totalRewards, only amount is transferred, while if amount > totalRewards, totalRewards is transferred

Reply:
See my correction of the last example you added above. `totalRewards` would there be larger than the amount inputted payout out too many funds. In the example total rewards added 1000, then 80, then 50 = 1130, but amount was only 1100 and only the 1100 is removed from his accrued, but he is still receiving 1130.